plugins/inputs/mavlink/mavlink_utils.go

```diff
@@ -0,0 +1,45 @@
+package mavlink
+
+import (
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/bluenviron/gomavlib/v3"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/filter"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/metric"
+)
+
+// Convert a Mavlink event into a struct containing Metric data.
+func convertEventFrameToMetric(frm *gomavlib.EventFrame, msgFilter filter.Filter) telegraf.Metric {
```

Comment:
I suggest moving this function to `mavlink.go` and get rid of the extra file. What do you think?